### DISCUSSION TITLE
Lottie Animation Not Loading, Causing App Crashes/Freezing for Some Users.

### DISCUSSION THREAD

**Discussion by MallaGnanadeep** [airbnb/lottie-ios#2501] [Q&A]
Hello everyone,

We are facing below issue for the Lottie animation we used.

Some users reported an issue where whenever they reach the screen where our Lottie animation file is shown, the Lottie animation file is not loaded. Along with this the app is either crashed or the screen is frozen.

However, while we debugged the same we found the below warning every time the Lottie animation is loaded. But we are unable to reproduce the scenario mentioned by the users.

> Encountered Core Animation compatibility issue while setting up animation:
> [Pre-comp 13.Pre-comp 12.Shape Layer 7.Trim Paths 1] The Core Animation rendering engine doesn't currently support applying trims to filled shapes (only stroked shapes).
> This animation may have additional compatibility issues, but animation setup was cancelled early to avoid wasted work.
> 
> Automatically falling back to Main Thread rendering engine. This fallback comes with some additional performance
> overhead, which can be reduced by manually specifying that this animation should always use the Main Thread engine.

This warning is shown every time we reach the screen. So we suspected the scenario reported by the users might be because of this warning.

Below are the details related

> **Lottie SDK version**  4.4.3
> 
> **Xcode version** 15.4
> 
> Below is the Lottie File used
> 
> [lottie.json](https://github.com/user-attachments/files/17817205/lottie.json)

Could you please help us with this issue?

Thank you very much for your help.

Best regards

**Comment by calda** [comment]
The console log from Lottie is expected and wouldn't be related to any crash or stability issue. 

The animation you shared loads and plays correctly in a sample app. For us to investigate a crash we'd need a sample project with repro steps demonstrating the issue.
